source/dub/dub.d

```diff
@@ -458,12 +468,25 @@ class Dub {
 			recipe.buildSettings.targetType = TargetType.executable;
 
 		auto pack = new Package(recipe, path.parentPath, null, "~master");
-		loadPackage(pack);
+		pack.m_infoFile = path;
+
+		return pack;
 	}
-	/// ditto
-	void loadSingleFilePackage(string path)
+
+	unittest
 	{
-		loadSingleFilePackage(NativePath(path));
+		enum filePath = NativePath("foo.d");
+
+		enum content = q{
+			/+ dub.sdl:
+				name "foo"
+			+/
+		};
+
+		auto dub = new Dub(".", null, SkipPackageSuppliers.all);
+		const pack = dub.loadSingleFilePackage(filePath, content);
```

Comment:
This fails when tested with Vibe.d:

```
Running ./bin/dub-test-library-nonet 
object.Exception@../.dub/packages/vibe-core-1.6.0/vibe-core/source/vibe/core/path.d(481): Path has no parent path
----------------
../.dub/packages/vibe-core-1.6.0/vibe-core/source/vibe/core/path.d:482 const @nogc @safe vibe.core.path.GenericPath!(vibe.core.path.PosixPathFormat).GenericPath vibe.core.path.GenericPath!(vibe.core.path.PosixPathFormat).GenericPath.parentPath() [0xb3168a]
source/dub/dub.d:470 dub.package_.Package dub.dub.Dub.loadSingleFilePackage(vibe.core.path.GenericPath!(vibe.core.path.PosixPathFormat).GenericPath, immutable(char)[]) [0xa6e163]
source/dub/dub.d:487 void dub.dub.Dub.__unittest_L476_C2() [0xa6e2d3]
```

The problem is that parentPath requires the path to have a parent. An easy fix would be to use `foo/bar` as dummy test path as `loadSingleFilePackage` can only be called with an absolute path.